samples/todo-app/database-api/MongoHelper.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace DatabaseApi
+{
+    public static class MongoHelper
+    {
+        private static string _connectionString { get; set; }
+
+        private static string _database { get; set; }
+
+        private static string _collection { get; set; }
+
+        private static MongoClient _mongoClient;
+
+        public static void Initialize(string connectionString, string database, string collection)
+        {
+            Console.WriteLine("MongoHelper initialization start");
+            _database = database;
+            _collection = collection;
+            _mongoClient = new MongoClient(connectionString);
+            ValidateMongoConnection(_mongoClient, _database);
+            Console.WriteLine("MongoHelper initialization end");
+        }
+
+        private static async void ValidateMongoConnection(MongoClient mongoClient, string database)
+        {
+            var db = mongoClient.GetDatabase(database);
+
+            while (true)
+            {
+                try
+                {
+                    BsonDocument isMongoLive = await db.RunCommandAsync((Command<BsonDocument>)"{ping:1}");
+                    if (!isMongoLive.Contains("ok") || isMongoLive["ok"].AsDouble != 1.0)
+                    {
+                        Console.Error.WriteLine("Mongo connection dead! Shutting down.");
+                        Environment.Exit(1);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Console.Error.WriteLine("Exception pinging Mongo:");
+                    Console.Error.WriteLine(e.ToString());
+                    Environment.Exit(1);
+                }
+
+                await Task.Delay(3000);
+            }
+        }
+
+        public static IEnumerable<TodoTask> GetTasks()
+        {
+            Console.WriteLine("Getting all TODO tasks");
+            var db = _mongoClient.GetDatabase(_database);
+            var collection = db.GetCollection<TodoTask>(_collection).WithWriteConcern(new WriteConcern("majority"));
+            return collection.AsQueryable();
+        }
+
+        public static async Task<TodoTask> CreateTask(TodoTask task)
+        {
+            Console.WriteLine($"Creating TODO task '{task.Title}'");
+            var db = _mongoClient.GetDatabase(_database);
+            var collection = db.GetCollection<TodoTask>(_collection).WithWriteConcern(new WriteConcern("majority"));
+            await collection.InsertOneAsync(task);
+            return task;
+        }
+
+        public static async Task<IEnumerable<TodoTask>> UpdateTask(string id, TodoTask updatedTask)
+        {
+            var tasks = GetTasks();
+            var task = tasks.First(task => task.Id == id);
```

Comment:
validation for the invalid id that is not present in the db.

Reply:
First would have thrown an exception if the task wasn't present, but I agree such exception wouldn't be very user friendly. I replaced it by a better exception.